cli/cmd/resource_run_cmd.go

```diff
@@ -83,23 +56,37 @@ func init() {
 	rootCmd.AddCommand(runCmd)
 }
 
-func getResourceType(runParams *runParameters, resourceParams *resourceParameters) (string, error) {
-	if resourceParams.ResourceName != "" {
-		return resourceParams.ResourceName, nil
+func runSingleFile(ctx context.Context) (string, error) {
+	orchestrator := runner.Orchestrator(
+		cliLogger,
+		config.GetAPIClient(cliConfig),
+		variableSetClient,
+		runnerRegistry,
+	)
+
+	if runParams.EnvID != "" {
+		runParams.VarsID = runParams.EnvID
 	}
 
-	if runParams.DefinitionFile != "" {
-		filePath := runParams.DefinitionFile
-		f, err := fileutil.Read(filePath)
-		if err != nil {
-			return "", fmt.Errorf("cannot read file %s: %w", filePath, err)
-		}
+	var definitionFile string
+	if len(runParams.DefinitionFiles) > 0 {
+		definitionFile = runParams.DefinitionFiles[0]
+	}
 
-		return strings.ToLower(f.Type()), nil
+	runParams := runner.RunOptions{
+		ID:              runParams.ID,
+		DefinitionFile:  definitionFile,
+		VarsID:          runParams.VarsID,
+		SkipResultWait:  runParams.SkipResultWait,
+		JUnitOuptutFile: runParams.JUnitOuptutFile,
+		RequiredGates:   runParams.RequiredGates,
 	}
 
-	return "", fmt.Errorf("resourceName is empty and no definition file provided")
+	exitCode, err := orchestrator.Run(ctx, runParams, output)
+	ExitCLI(exitCode)
 
+	// ExitCLI will exit the process, so this return is just to satisfy the compiler
+	return "", err
```

Comment:
I remember this comic every time we need to satisfy our Go compiler xD :
![image](https://github.com/kubeshop/tracetest/assets/12397879/4f4b5c61-c063-4e31-a462-84f3e63b9b3a)